includes/class-scriptlesssocialsharing-output.php

```diff
@@ -147,6 +147,23 @@ public function do_buttons( $output, $heading = true ) {
 	}
 
 	/**
+	 * Get the permalink to be shared via the button.
+	 *
+	 * @param  string $button_name The name of the button, e.g. 'twitter', 'facebook'.
+	 * @return string The URL to be shared.
+	 */
+	protected function get_permalink( string $button_name ) {
+		$attributes = $this->attributes();
+		return rawurlencode(
```

Comment:
I don't think it does make sense to move it to line 140.

On [line 140](https://github.com/robincornett/scriptless-social-sharing/blob/develop/includes/class-scriptlesssocialsharing-output.php#L140)

the `$url` variable contains the full URL
```
https://twitter.com/intent/tweet?text=test&#038;url=http%3A%2F%2Fsalferrarello.dev%2Ftest%2F%3Futm_source%3Dsalcodecom%26utm_medium%3Dblogpost&#038;via=salcode&#038;related=salcode
```

We only want to `rawurlencode()` the URL that is being passed as a parameter, i.e.

```
http://salferrarello.dev/test/?utm_source=salcodecom&amp;utm_medium=blogpost&amp;via=salcode&amp;related=salcode
```